src/r2x/utils.py

```diff
@@ -181,11 +181,34 @@ def update_dict(base_dict: dict, override_dict: ChainMap | dict | None = None) -
     return base_dict
 
 
-def read_yaml(fname: str) -> dict:
+def read_user_dict(fname: str) -> dict:
     """Load Yaml to Python dictionary."""
-    # Read configuration file
-    with open(fname) as f:
-        return yaml.safe_load(f)
+    if fname.strip().startswith("{") or fname.strip().startswith("["):
```

Comment:
if fname starts with "[", then the return type will be a list, we should either change the return type or only allow input with "{"

Reply:
Good call. I have mixed feelings about this. Shall we just remove the option to accept a JSON as a string?

Reply:
I think the string option is nice for flexibility, but considering the variable is a user_dict, we can probably raise an InputError or some other exception if the user_dict starts with "[".

Reply:
Ok, we raise a ValueError now.